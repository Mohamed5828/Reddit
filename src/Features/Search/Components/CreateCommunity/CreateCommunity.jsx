import { ContainerCreateComm } from "./CreateCommunity.styled";
import BackgroundImage from "../../Assets/search-results-community-banner.png";
import logo from "../../Assets/snoo-thinking.png";

/**
 * Component that contains the CreateCommunity component.
 *
 * @Component
 * @returns {React.Component}
 */
const CreateCommunity = () => {
  return (
    <ContainerCreateComm>
      <div className="create">
        <div className="content">
          <img src={BackgroundImage} alt="" className="first" />
          <img src={logo} alt="" className="second" />
          <div className="txt">Have an idea for a new community?</div>
        </div>
        <div className="btn-create">
          <button>Create Community</button>
        </div>
      </div>
    </ContainerCreateComm>
  );
};

export default CreateCommunity;
