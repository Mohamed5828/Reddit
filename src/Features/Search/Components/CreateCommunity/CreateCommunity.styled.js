import styled from "styled-components";
export const ContainerCreateComm = styled.div`
  .create {
    background: ${({ theme }) => theme.background.primary};
    border: thin solid #ccc;
    border-radius: 4px;
    box-shadow: 0 2px 4px 0 rgb(0 0 0 / 6%);
    margin-bottom: 16px;
    max-width: 100%;
  }
  @media (max-width: 500px) {
    display: none;
  }
  .create .content {
    background: ${({ theme }) => theme.background.primary};
    border-radius: 4px;
    box-sizing: border-box;
    color: ${({ theme }) => theme.color.titleText_of_link};
    overflow: hidden;
    position: relative;
  }
  .create img.first {
    height: 120px;
    width: 100%;
  }
  .create img.second {
    height: 126px;
    width: 70px;
    left: 50%;
    position: absolute;
    top: 57px;
    transform: translateX(-50%);
  }
  .txt {
    font-family: Noto Sans, Arial, sans-serif;
    font-size: 14px;
    font-weight: 400;
    line-height: 21px;
    margin-top: 70px;
    text-align: center;
    color: ${({ theme }) => theme.color.titleText_of_link};
  }
  .btn-create {
    padding: 12px;
  }
  button {
    width: 100%;
    position: relative;
    background-color: ${({ theme }) => theme.background.fill};
    /* border: 1px solid transparent; */
    color: ${({ theme }) => theme.color.secondary};

    font-family: Noto Sans, Arial, sans-serif;
    font-size: 14px;
    font-weight: 700;
    letter-spacing: unset;
    line-height: 17px;
    text-transform: unset;
    min-height: 32px;
    min-width: 32px;
    padding: 4px 16px;
    align-items: center;
    border-radius: 9999px;
    box-sizing: border-box;
    display: flex;
    justify-content: center;
    text-align: center;
    position: relative;
    border: 1px solid ${({ theme }) => theme.color.secondary};
  }
  .btn-create button {
  }
`;
