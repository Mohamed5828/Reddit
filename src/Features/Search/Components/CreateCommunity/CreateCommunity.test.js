import { shallow } from "enzyme";
import CreateCommunity from "./CreateCommunity";
describe("Create Community", () => {
  it("this is a test for Create Community", () => {
    expect(shallow(<CreateCommunity />)).toMatchSnapshot();
  });
});
