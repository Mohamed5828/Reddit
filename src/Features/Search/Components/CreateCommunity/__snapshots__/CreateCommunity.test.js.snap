// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Community this is a test for Create Community 1`] = `
<styled.div>
  <div
    className="create"
  >
    <div
      className="content"
    >
      <img
        alt=""
        className="first"
        src="search-results-community-banner.png"
      />
      <img
        alt=""
        className="second"
        src="snoo-thinking.png"
      />
      <div
        className="txt"
      >
        Have an idea for a new community?
      </div>
    </div>
    <div
      className="btn-create"
    >
      <button>
        Create Community
      </button>
    </div>
  </div>
</styled.div>
`;
